scripts/exgfs_atmos_postsnd.sh

```diff
@@ -44,47 +49,116 @@ export NINT3=${FHOUT_GFS:-3}
 
 rm -f -r "${COM_ATMOS_BUFR}"
 mkdir -p "${COM_ATMOS_BUFR}"
+export COM_ATMOS_BUFR="${COM_ATMOS_BUFR}"
+
 GETDIM="${USHgfs}/getncdimlen"
-LEVS=$(${GETDIM} "${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atmf000.${atmfm}" pfull)
+export LEVS=$(${GETDIM} "${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atmf000.${atmfm}" pfull)
 declare -x LEVS
 
-### Loop for the hour and wait for the sigma and surface flux file:
-export FSTART=$STARTHOUR
+### wait for the sigma and surface flux file:
 sleep_interval=10
 max_tries=360
 #
-while [ $FSTART -lt $ENDHOUR ]
-do
-export FINT=$NINT1
-   # Define the end hour for the input
-   export FEND=$(expr $FSTART + $INCREMENT) 
-   if test $FEND -lt 100; then FEND=0$FEND; fi 
-   if [ $FSTART -eq 00 ]
-   then 
-       export F00FLAG=YES
-   else
-       export F00FLAG=NO
-   fi
-   
-   if [ $FEND -eq $ENDHOUR ]
-   then
-       export MAKEBUFR=YES
-   fi
-
-   filename="${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atm.logf${FEND}.${logfm}"
-   if ! wait_for_file "${filename}" "${sleep_interval}" "${max_tries}"; then
-     err_exit "FATAL ERROR: logf${FEND} not found after waiting $((sleep_interval * ( max_tries - 1) )) secs"
-   fi
 
-## 1-hourly output before $NEND1, 3-hourly output after
-   if [[ $((10#$FEND)) -gt $((10#$NEND1)) ]]; then
-     export FINT=$NINT3
-   fi
-   ${USHgfs}/gfs_bufr.sh
-  
-   export FSTART="${FEND}"
+# Initialize an empty list to store the hours
+hour_list=()
+
+# Generate hours from 0 to NEND1 with interval NINT1
+# Convert ENDHOUR to decimal through $((10#$ENDHOUR)) to avoid it is thought as octal number
+for (( hour=0; hour<=$((10#$NEND1)) && hour<=$((10#$ENDHOUR)); hour+=$((10#$NINT1)) )); do
```

Comment:
When I ran the test, I mistakenly set ENDHOUR=012 instead of 12. This typo caused an issue with the job running, so I added #10 as a safeguard. Anyway, I can remove it if it causes confusion.